### PR TITLE
Revert of PR #4712, restores text wrapping behavior

### LINKED ISSUE #5081: Difference with text wrapping betwen 1.1.9 -> 1.2.0 and 1.3.0
### How would this new feature help [increase access](https://github.com/processing/p5.js/blob/main/contributor_docs/access.md) to p5.js?
I believe that this feature would help allow the text function to do what the programmer wants it to do and relieve programmers from confusion about differences in p5 sketches, as I myself was making something along with a friend. I had made my sketch before version 1.2.0 was released and my friend after, we did very similar things and our code was producing different output and confused us for quite a while.

#### Most appropriate sub-area of p5.js?

- [ ] Accessibility (Web Accessibility)
- [ ] Build tools and processes
- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] DOM
- [ ] Events
- [ ] Friendly error system
- [ ] Image
- [ ] IO (Input/Output)
- [ ] Localization
- [ ] Math
- [ ] Unit Testing
- [x] Typography
- [ ] Utilities
- [ ] WebGL
- [ ] Other (specify if possible)

#### Feature enhancement details:

There is a difference between p5's text function when using 5 parameters to create a textbox so that the text wraps along multiple lines. 

In p5 version 1.1.9 the text wrapping works on a word by word basis (will always create a line break on a whitespace and not in the middle of a word), example [here](https://editor.p5js.org/Gabe_W/sketches/fDPm7YSJJ). 

In versions 1.2.0 and 1.3.0 this functionality changed and now the text wrapping will split in the middle of the text, creating a hyphen on the end of the line, examples [here ](https://editor.p5js.org/Gabe_W/sketches/MiqRHHiQD) for 1.2.0 and [here ](https://editor.p5js.org/Gabe_W/sketches/JSXg362bI) for 1.3.0.

My request for feature enhancement would be a way to switch between these two modes of text wrapping. Perhaps add another parameter to the text function? This seems like a kind of clunky solution but at the moment I can't think of anything better. Does this feature already exist? If so how would I switch between the modes? Thanks for your consideration!

### PR DESCRIPTION
**Pull request by lawreka**
Resolves #5081

 Changes:
Reverts PR #4712 and restores text wrapping behavior as seen in version 1.1.9.

 Screenshots of the change:
Current rendering of 'Lorem ipsum dolor sit amet, consectetur adipiscing elit' in 100, 100 canvas:
Screenshot 2021-03-14 at 21 22 32
Fixed:
Screenshot 2021-03-14 at 21 19 50

Note this also re-introduces Issue #4652, future improvements should change how single words that exceed the length of a line can be handled by hyphenation or not.  

#### PR Checklist

- [x] `npm run lint` passes

Inline documentation and unit tests unchanged

[Inline documentation]: https://github.com/processing/p5.js/blob/main/contributor_docs/inline_documentation.md
[Unit tests]: https://github.com/processing/p5.js/tree/main/contributor_docs#unit-tests

### CODE REVIEW

**Review by outofambit — APPROVED** [review]
looks good to me! thank you @lawreka!!

**Comment by outofambit** [comment]
@all-contributors please add @lawreka for code and ideas

File: src/core/p5.Renderer.js [hunk 1]
```diff
@@ -216,10 +216,10 @@ p5.Renderer.prototype.text = function(str, x, y, maxWidth, maxHeight) {
   let jj;
   let line;
   let testLine;
+  let currentLineLength;
   let testWidth;
   let words;
   let totalHeight;
-  let shiftedY;
```
Comment (lawreka) [inline src/core/p5.Renderer.js:222]: thanks for your review @outofambit, but I think I made a mistake here... adding this shiftedY variable was part of the original PR that I'm reverting part of, but doesn't have to do with my fix, so I'm just going to push a small change to not touch this before this gets merged :)

Comment (outofambit) [reply src/core/p5.Renderer.js:222]: tbh i just did a line by line comparison of this against the original PR :)

thanks for catching that!

Comment (lawreka) [reply src/core/p5.Renderer.js]: I did the same and didn't see it until after your approval either 😅 just pushed the fix, should be good 2 go now

**Pull request merged**